src/components/CalendarLayout/_components/CalendarList/CalendarList.tsx

```diff
@@ -1,80 +1,159 @@
-import { MagnifyingGlass as SearchIcon } from '@phosphor-icons/react'
-import { Flex, Input, List, Modal, Space } from 'antd'
+import {
+	DotsThreeVertical as MoreIcon,
+	PencilSimple as PencilSimpleIcon,
+	MagnifyingGlass as SearchIcon,
+} from '@phosphor-icons/react'
+import { Badge, Dropdown, Flex, Input, List, Segmented, Space } from 'antd'
+import { ItemType } from 'antd/lib/menu/hooks/useItems'
+import dayjs from 'dayjs'
 import 'dayjs/locale/fr'
 import { useState } from 'react'
+import { useNavigate } from 'react-router-dom'
 
-import { Appointment, appointments } from '../appointment'
+import { IconButton } from '@components'
 
+import { ICalendarListProps } from './CalendarList-types'
+
+import '../../../FavoritesList/FavoritesList-styles.less'
 import './CalendarList-styles.less'
 
-export function CalendarList() {
+export function CalendarList(props: ICalendarListProps) {
+	const { events, appointments } = props
 	const [search, setSearch] = useState('')
-	const [isModalVisible, setIsModalVisible] = useState(false)
-	const [selectedAppointment, setSelectedAppointment] = useState<Appointment | null>(null)
+	const [selectedButton, setSelectedButton] = useState('Rendez-vous')
+	const navigate = useNavigate()
+	const currentDate = dayjs()
 
-	const showAppointmentDetails = (appointment: Appointment) => {
-		setSelectedAppointment(appointment)
-		setIsModalVisible(true)
-	}
-
-	const handleCancel = () => {
-		setIsModalVisible(false)
-	}
+	const sortedAndFilteredEvents = events
+		.filter((event) => dayjs(event.event_date).isAfter(currentDate))
+		.sort((a, b) => dayjs(a.event_date).valueOf() - dayjs(b.event_date).valueOf())
 
+	const sortedAndFilteredAppointments = appointments
+		.filter((appointment) => dayjs(appointment.planned_date).isAfter(currentDate))
+		.sort((a, b) => dayjs(a.planned_date).valueOf() - dayjs(b.planned_date).valueOf())
 	const handleSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
 		setSearch(e.target.value)
 	}
 
-	const filteredAppointments = appointments.filter((appointment) => {
-		const matchName = appointment.name.toLowerCase().includes(search.toLowerCase())
-		const matchDescription = appointment.description.toLowerCase().includes(search.toLowerCase())
-		const matchPosition = appointment.position.toLowerCase().includes(search.toLowerCase())
+	const handleSegmentChange = (value: string | number) => {
+		if (typeof value === 'string') {
+			setSelectedButton(value)
+		}
+	}
+
+	const getMenuAppointments = (id: number): ItemType[] => [
+		{
+			key: 'create-follow-up',
+			label: 'Modifier le rendez-vous',
+			icon: <PencilSimpleIcon size={16} weight="bold" />,
+			onClick: () => navigate(`/appointments?action=edit&id=${id}`),
+		},
+	]
+
+	const getMenuEvents = (id: string): ItemType[] => [
```

Comment:
Renomme la fonction, et ajoute un underscore à l'id pour éviter les problèmes de lint :
```suggestion
	const getEventsMenu = (_id: string): ItemType[] => [
```